Dockerfile

```diff
@@ -0,0 +1,28 @@
+FROM registry.redhat.io/ubi8:latest as builder
+
+# Install required RPMs and ensure that the packages were installed
+USER root
+RUN yum --setopt=tsflags=nodocs install -y unzip tar rsync shadow-utils \
+    && rpm -q unzip tar rsync shadow-utils
+
+
+# Install required RPMs and ensure that the packages were installed
+USER root
+RUN yum --setopt=tsflags=nodocs install -y java-11-openjdk-devel \
+    && rpm -q java-11-openjdk-devel
+
+# Install required RPMs and ensure that the packages were installed
+USER root
+RUN yum --setopt=tsflags=nodocs install -y maven \
+    && rpm -q maven
```

Comment:
So - I know this is a builder image, but good practice is usually installing all packages in one command or at the very least in the same layer, since all of these layers are pretty static.

```suggestion
RUN yum --setopt=tsflags=nodocs install -y \
    # required RPMs for unpackaging jar files
    unzip tar rsync shadow-utils \
    # install java + maven
    java-11-openjdk-devel  maven
```

the `rpm -q` at the end isn't quite necessary either in my opinion - if the installation fails dnf will fail and exit the build with error.